core/templates/pages/exploration-editor-page/editor-tab/state-version-history/state-version-history.component.ts

```diff
@@ -59,8 +59,12 @@ export class StateVersionHistoryComponent {
       this.versionHistoryService.getBackwardStateDiffData());
     modalRef.componentInstance.newState = stateDiffData.newState;
     modalRef.componentInstance.oldState = stateDiffData.oldState;
-    modalRef.componentInstance.newStateName = stateDiffData.newState.name;
-    modalRef.componentInstance.oldStateName = stateDiffData.oldState.name;
+    if (stateDiffData.newState && stateDiffData.newState.name) {
```

Comment:
This needs some explanation as to why newState can be null/undefined (I'm not sure that that makes sense to me conceptually) or why it's possible for it not to have a name. Ditto below.

Reply:
Added some explanation via comments. Please take a look.

Reply:
Er, I don't see any comments?

Reply:
I have changed this block of code now. PTAL. Thanks!